stixpy/coordinates/frames.py

```diff
@@ -0,0 +1,160 @@
+import astropy
+import astropy.coordinates as coord
+import astropy.units as u
+from astropy.coordinates import QuantityAttribute
+from astropy.wcs import WCS
+from sunpy.coordinates.frameattributes import ObserverCoordinateAttribute
+from sunpy.coordinates.frames import HeliographicStonyhurst, SunPyBaseCoordinateFrame
+from sunpy.sun.constants import radius as _RSUN
+
+from stixpy.net.client import STIXClient  # noqa
+from stixpy.utils.logging import get_logger
+
+logger = get_logger(__name__, "DEBUG")
+
+__all__ = ["STIXImaging"]
+
+STIX_X_SHIFT = 26.1 * u.arcsec  # fall back to this when non sas solution available
+STIX_Y_SHIFT = 58.2 * u.arcsec  # fall back to this when non sas solution available
+STIX_X_OFFSET = 60.0 * u.arcsec  # remaining offset after SAS solution
+STIX_Y_OFFSET = 8.0 * u.arcsec  # remaining offset after SAS solution
+
+
+class STIXImaging(SunPyBaseCoordinateFrame):
+    r"""
+    STIX Imaging Frame
+
+    - ``x`` (aka "theta_x") along the pixel rows (e.g. 0, 1, 2, 3; 4, 5, 6, 8).
+    - ``y`` (aka "theta_y") along the pixel columns (e.g. A, B, C, D).
+    - ``distance`` is the Sun-observer distance.
+
+    Aligned with SIIX 'pixels' +X corresponds direction along pixel row toward pixels
+    0, 4 and +Y corresponds direction along columns towards pixels 0, 1, 2, 3.
+
+    .. code-block:: text
+
+      Col\Row
+            ---------
+           | 3  | 7  |
+        D  |   _|_   |
+           |  |11 |  |
+            ---------
+           | 2  | 6  |
+        C  |   _|_   |
+           |  |10 |  |
+            ---------
+           | 1  | 5  |
+        B  |   _|_   |
+           |  | 9 |  |
+            ---------
+           | 0  | 4  |
+        A  |   _|_   |
+           |  | 8 |  |
+            ---------
+
+    """
+    observer = ObserverCoordinateAttribute(HeliographicStonyhurst)
+    rsun = QuantityAttribute(default=_RSUN, unit=u.km)
+
+    frame_specific_representation_info = {
+        coord.SphericalRepresentation: [
+            coord.RepresentationMapping("lon", "Tx", u.arcsec),
+            coord.RepresentationMapping("lat", "Ty", u.arcsec),
+            coord.RepresentationMapping("distance", "distance"),
+        ],
+        coord.UnitSphericalRepresentation: [coord.RepresentationMapping('lon', 'Tx', u.arcsec),
+                                            coord.RepresentationMapping('lat', 'Ty', u.arcsec)]
+    }
+
+
+def stix_wcs_to_frame(wcs):
+    r"""
+    This function registers the coordinate frames to their FITS-WCS coordinate
+    type values in the `astropy.wcs.utils.wcs_to_celestial_frame` registry.
+
+    Parameters
+    ----------
+    wcs : astropy.wcs.WCS
+
+    Returns
+    -------
+    astropy.coordinates.BaseCoordinateFrame
+    """
+    if hasattr(wcs, "coordinate_frame"):
+        return wcs.coordinate_frame
+
+    # Not a STIX wcs bail out early
+    if set(wcs.wcs.ctype) != {'SXLN-TAN', 'SXLT-TAN'}:
```

Comment:
Perhaps `{'SXLN-TAN', 'SXLT-TAN'}` should be stored as a module-level attribute and accessed throughout the package from there.  That would allow it can be changed easily in the future without a million find/replace.